File: backend/src/main/java/com/joying/chat/dto/PresenceUpdateEvent.java
package com.joying.chat.dto;

import java.time.Instant;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 접속 상태가 바뀐 것을 알리는 이벤트.
 */
@Getter
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class PresenceUpdateEvent {

	private Long memberId;

	@JsonProperty("isOnline")
	private boolean isOnline;

	private Instant lastSeenAt;
}

File: backend/src/main/kotlin/com/joying/chat/dto/PresenceUpdateEvent.kt
package com.joying.chat.dto

import java.time.Instant

/**
 * 온라인 상태 변경 이벤트
 *
 * WebSocket으로 전송하여 채팅방 목록의 온라인 상태를 실시간으로 업데이트
 */
data class PresenceUpdateEvent(
    val memberId: Long,
    val isOnline: Boolean,
    val lastSeenAt: Instant? = null
)